exonum/src/events/network.rs

```diff
@@ -229,7 +229,7 @@ impl NetworkHandler {
                 let holder = incoming_connections_counter.clone();
                 // Check incoming connections count
                 let connections_count = Rc::strong_count(&incoming_connections_counter) - 1;
-                if connections_count > incoming_connections_limit {
+                if connections_count >= incoming_connections_limit {
```

Comment:
An alternative would be

```diff
-                let connections_count = Rc::strong_count(&incoming_connections_counter) - 1;
+                let connections_count = Rc::strong_count(&incoming_connections_counter);
                 if connections_count > incoming_connections_limit {
```

which is arguably more simple, but the current form is fine too.

The current decremented form even has its merits: `connection_count` really reflects the connection count, not including that additional reference held by the server's future.